File: backend/Models/SalesOrder.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("sales_orders")]
    public class SalesOrder
    {
        [Key] public int Id { get; set; }

        public int? CustomerId { get; set; }

        [ForeignKey("CustomerId")]
        public Customer? Customer { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.Now;

        public decimal TotalAmount { get; set; } = 0;

        [StringLength(20)]
        public string Status { get; set; } = "pending";

        public ICollection<SalesOrderItem>? Items { get; set; }
    }
}

File: backend/src/main/java/pika/hungt1/dx/entity/SalesOrder.java
package pika.hungt1.dx.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;
import java.time.Instant;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name = "sales_orders")
@Data @NoArgsConstructor @AllArgsConstructor
public class SalesOrder {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name="customer_id")
    private Customer customer;

    @Column(name="order_date")
    private Instant orderDate = Instant.now();

    @Column(name="total_amount")
    private BigDecimal totalAmount = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    private Status status = Status.pending;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SalesOrderItem> items;

    public enum Status { pending, paid, cancelled }
}
